ci/jenkins/jobs/macros.yaml

```diff
@@ -293,3 +293,52 @@
           set -e
           DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
           ./ci/jenkins/test.sh --testcase '{conformance_type}' --registry ${{DOCKER_REGISTRY}} --testbed-type jumper
+
+- builder:
+    name: builder-capa-matrix
+    builders:
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
+          export JOB_NAME="matrix-${TEST_OS}-k8s-${K8S_VERSION//./-}-build-num"
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --setup-only --provider aws --aws-region "us-west-2" --aws-access-key-id "${AWS_ACCESS_KEY}" --aws-secret-access-key "${AWS_SECRET_KEY}" --aws-service-user-name "${AWS_SERVICE_USER_NAME}" --aws-service-user-role "${AWS_SERVICE_USER_ROLE_ARN}" --aws-vpc-id "${CAPA_VPC}" --aws-subnet-id "${CAPA_SUBNET}" 
+          testcases=("e2e" "conformance" "networkpolicy")
+          for testcase in "${testcases[@]}"; do
+            ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --testcase "$testcase" --test-only --provider aws --aws-region "us-west-2" --aws-access-key-id "${AWS_ACCESS_KEY}" --aws-secret-access-key "${AWS_SECRET_KEY}" --aws-service-user-name "${AWS_SERVICE_USER_NAME}" --aws-service-user-role "${AWS_SERVICE_USER_ROLE_ARN}" --aws-vpc-id "${CAPA_VPC}" --aws-subnet-id "${CAPA_SUBNET}"
+            test_result=$?
+            echo "$test_result" > "test-$testcase-rc.txt"
+          done
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --cleanup-only --provider aws
+          failure=0
+          for result in test-*-rc.txt; do
+            r="$(cat $result)"
+            if [ "$r" -ne 0 ]; then
+              echo Test Failed: $result : $r
+              failure=failure+1
+            fi
+          done
+          exit failure
+
+- builder:
+    name: builder-capa-aws-clean-up
+    builders:
+      - shell: |-
+          #!/bin/bash
+          for clustername in `kubectl get clusters -A -l jenkins-test --no-headers | grep test | awk '{print $1}'`; do 
+              echo $clustername;
+              timestamp="$(kubectl get clusters -o jsonpath="{.metadata.creationTimestamp}" $clustername -n $clustername)"
+              t1=$(date -d "$timestamp" "+%s")
+              now=$( date "+%s")
+              delta=$((now-t1))
+              if [ "$delta" -gt "14400" ]; then 
+                timeout 2m echo ./ci/jenkins/test-vmc.sh --cluster-name "$clustername" --cleanup-only --provider 
+                result=$?
+                if [[ $result == 124 ]]; then
+                  echo "#error# clean up job of ${clustername} time out"
+                fi
+                if [[ $result == 124 ]]; then
+                  echo "delete old job successfully"
+                fi
+              fi
```

Comment:
can handle failure messages in more hierarchical manner along with descriptive messages something for example
# Check if the cluster is older than 4 hours (14,400 seconds)
    if [ "$delta" -gt "14400" ]; then 
        # Run the cleanup command with a timeout of 2 minutes
        timeout 2m ./ci/jenkins/test-vmc.sh --cluster-name "$clustername" --cleanup-only --provider aws
        result=$?
        
        # Handle cleanup result
        if [ $result -eq 124 ]; then
            echo "Cleanup job for $clustername timed out"
        elif [ $result -eq 0 ]; then
            echo "Deleted old cluster successfully: $clustername"
        else
            echo "Cleanup job for $clustername failed with exit code $result"
        fi
    else
        echo "Cluster $clustername is not older than 4 hours, skipping cleanup"
    fi